### PR TITLE
build gradle plugin to 0.6.0

### PR DESCRIPTION
**Pull request by proppy**
Staging this for testing, we should only merge when the `-alpha` suffix is dropped.

### CODE REVIEW

**Comment by proppy** [comment]
It seems that `compileOptions.with` should be in the `android {} block`

**Comment by proppy** [comment]
rebased.

File: MoreTeapots/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.8-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.9-all.zip
```
Comment (rschiu) [inline MoreTeapots/gradle/wrapper/gradle-wrapper.properties]: Need 2.10 now.  Do you want to keep -bin or switch to -all?

Comment (proppy) [inline MoreTeapots/gradle/wrapper/gradle-wrapper.properties]: I think if we keep `-bin` android studio produce a warning/quickfix asking to switch to `-all`?

Comment (rschiu) [inline MoreTeapots/gradle/wrapper/gradle-wrapper.properties]: That's true.  Let's switch to -all then.

**Comment by proppy** [comment]
@rschiu FYI, build failed for `Teapot` and `MoreTeapot` with the following error:
```
> Building 85% > :test    * What went wrong:
> Building 85% > :test    A problem was found with the configuration of task ':app:stripSymbolsArm64-v8aDebugAllSharedLibrary'.
> Building 85% > :test    > 
> Building 85% > :test    Directory '/home/travis/build/googlesamples/android-ndk/MoreTeapots/app/build/intermediates/binaries/debug/all/obj/arm64-v8a' specified for property 'inputFolder' does not exist.
```

**Comment by rschiu** [comment]
Please file a bug for me.

**Comment by proppy** [comment]
@rschiu done: https://code.google.com/p/android/issues/detail?id=200078&thanks=200078&ts=1454368786

**Comment by proppy** [comment]
Sample have already been updated to 0.7.0

**Pull request closed without merging**